### PR TITLE
show_node displays connected networks

### LINKED ISSUE #499: show_node() should display attached networks
Currently, when attempting to detach a node from a project, one can encounter errors due to the node still being connected to networks:
```
$ haas project_detach_node moc-prod node-123
Unexpected status code: 409
Response text:
{"msg": "Node attached to a network", "type": "BlockedError"}
```

Especially if this node were set up by someone else, it can be difficult to figure out which networks it is connected to.

show_node() should display a list of such networks, as well as the channels on which they are connected. This could be implemented as a show_nic() command, which is then called within show_node() for each nic.

This is somewhat the inverse of #496.

This was encountered by several folks, including @rahulait and @weinbe2

### PR DESCRIPTION
**Pull request by kylehogan**
changes to api.py, unit tests for api.py, and rest_api.md from #469 by @zenhack 
closes #499

### CODE REVIEW

**Comment by henn** [comment]
I suggest taking a look at the rendered [rest_api.md output](https://github.com/kylehogan/haas/blob/show_networks_of_node/docs/rest_api.md) to check the syntax.

You can also install the python-markdown package to render it locally before committing.

File: docs/rest_api.md [hunk 1]
```diff
@@ -314,25 +314,25 @@ Response body:
...
-    {
-        "name": "box02",
-        "free": true,
-        "nics": ["ipmi", "pxe", "external",...]
-    }
-
+Show details of a node.
+
+Returns a JSON object representing a node.
+The object will have at least the following fields:
```
Comment (henn) [inline docs/rest_api.md]: Needs a newline after the :

File: docs/rest_api.md [hunk 2]
```diff
@@ -314,25 +314,25 @@ Response body:
...
+        * "name", the name/label of the node (string).
+        * "project", indicates whether the node belongs to a project, displaying None if not and the name of the project if so
+            to a project.
+        * "nics", a list of nics, each represted by a JSON object having
+            at least the following fields:
+                - "label", the nic's label.
+                - "macaddr", the nic's mac address.
+
+Example:  '{"name": "node1",
```
Comment (henn) [inline docs/rest_api.md]: Look at how `list_free_nodes()` and other JSON blobs are formatted

**Comment by SahilTikale** [comment]
Hi @kylehogan 
I suggest you also make the following change in api.py 

Instead of 
`'project': 'None' if node.project_id is None else node.project.label,`

Put instead
`'project': None if node.project_id is None else node.project.label,`

You will have to fix the corresponding `tests/unit/api.py` and `docs/rest_api.md`

**Comment by SahilTikale** [comment]
Look at the output of show_node when a node is free (not allocated to any project)
it might look like
`{"project": null, ..... }` 
fix the unit tests accordingly and it will start passing the unit tests.

**Comment by kylehogan** [comment]
tests passing with expect 
`{"project": None, ...}`
expecting the project to be null still failed test - output from Travis showed it comparing null to None

File: haas/api.py [hunk 3]
```diff
@@ -773,32 +773,22 @@ def list_project_networks(project):
...
-                - "macaddr", the nic's mac address.
-
-    Example:  '{"name": "node1",
-                "free": True,
-                "nics": [{"label": "nic1", "macaddr": "01:23:45:67:89"},
-                         {"label": "nic2", "macaddr": "12:34:56:78:90"}]
-               }'
-    """
-    
+    '''
```
Comment (henn) [inline haas/api.py]: Please follow [PEP257](https://www.python.org/dev/peps/pep-0257/) here, specifically:
 * Using double-quotes
 * Either using a one-liner, or following the conventions for where lines start/stop.

**Comment by henn** [comment]
@kylehogan and I discussed this, and agreed that it's best to try to keep the examples and documentation in rest_api.md so that there's one source of truth.

**Comment by henn** [comment]
@kylehogan @SahilTikale - do you guys think you could (respectively) make the small changes we discussed and review this by tomorrow? We are holding up merging #491 for this so that we can deploy the changes into production without DB issues.

Thanks

**Comment by kylehogan** [comment]
Which changes are you referring to?

**Comment by SahilTikale** [comment]
in response to 
> @kylehogan and I discussed afc6cbd, and agreed that it's best to try to keep the examples and documentation in rest_api.md so that there's one source of truth.

For the purpose of this pull request it does not matter, since the same information will be repeated between `rest_api.md` and `USING.rst`

We may want to update both this files when updating information where drivers involved. 
`rest_api.md` will capture generic body of the api-request
`USING.rst` will provide example of a specific type of implementation for the api request and along with usage for CLI arguments.

File: docs/rest_api.md [hunk 4]
```diff
@@ -314,25 +314,28 @@ Response body:
...
-* "nics", a list of nics, each represted by a JSON object having
-    at least the following fields:
-        * "label", the nic's label.
-        * "macaddr", the nic's mac address.
+Returns a JSON object representing a node.
+The object will have at least the following fields:
 
-Response body:
+        * "name", the name/label of the node (string).
+        * "project", indicates whether the node belongs to a project, displaying null if not and the name of the project if so.
```
Comment (SahilTikale) [inline docs/rest_api.md]: can you revise the sentence. 
Ending a sentence on 'so' does not sound right. 
the meaning is not clear as to when it is null.

File: haas/api.py [hunk 5]
```diff
@@ -773,32 +773,38 @@ def list_project_networks(project):
...
+    """Show the details of a node.
 
     Returns a JSON object representing a node.
+
     The object will have at least the following fields:
+
         * "name", the name/label of the node (string).
-        * "free", indicates whether the node is free or has been allocated
-            to a project.
+        * "project", indicates whether the node belongs to a project, displaying null if not and the name of the project if so.
```
Comment (SahilTikale) [inline haas/api.py]: sentence revision: same as before.

File: docs/rest_api.md [hunk 6]
```diff
@@ -314,25 +314,28 @@ Response body:
...
+            at least the following fields:
 
-    {
-        "name": "box02",
-        "free": true,
-        "nics": ["ipmi", "pxe", "external",...]
-    }
+                - "label", the nic's label.
+                - "macaddr", the nic's mac address.
+
```
Comment (SahilTikale) [inline docs/rest_api.md]: Information related to the json object that shows networks attached to nics is missing.

File: docs/rest_api.md [hunk 7]
```diff
@@ -314,25 +314,28 @@ Response body:
...
+Response body:
 
+	{"name": "node1",
+	 "project": "project1",
+         "nics": [{"label": "nic1", "macaddr": "01:23:45:67:89"},
+                  {"label": "nic2", "macaddr": "12:34:56:78:90"}]
+	}
+
```
Comment (SahilTikale) [inline docs/rest_api.md]: Is the response body missing information related to attached networks.

File: haas/api.py [hunk 8]
```diff
@@ -773,32 +773,38 @@ def list_project_networks(project):
...
-    Example:  '{"name": "node1",
-                "free": True,
-                "nics": [{"label": "nic1", "macaddr": "01:23:45:67:89"},
-                         {"label": "nic2", "macaddr": "12:34:56:78:90"}]
-               }'
+    Example: '{"name": "node1",
+              "project": "project1",
+              "nics": [{"label": "nic1", "macaddr": "01:23:45:67:89"},
+                       {"label": "nic2", "macaddr": "12:34:56:78:90"}]
+	      }'
```
Comment (SahilTikale) [inline haas/api.py]: should this example show information about networks attached ?

File: tests/unit/api.py [hunk 9]
```diff
@@ -1238,6 +1239,13 @@ def test_some_non_free_nodes(self, db):
         assert json.loads(api.list_free_nodes()) == ['master-control-program']
 
     def test_show_node(self, db):
+        """Test the show_node api call.
+
+        We create a node, and query it twice: once before it is reserved,
+        and once after it has been reserved by a project and attached to
+        a network. Two things should change: (1) "free" should be false,
+        and (2) the newly attached network should be listed.
```
Comment (SahilTikale) [inline tests/unit/api.py]: Update the doc-string  point (1) appropriately.

**Comment by SahilTikale** [comment]
@kylehogan 
can you write a test where a node with two nics is registered and  attached different networks,  
then `show_node` showing that information.

File: docs/rest_api.md [hunk 10]
```diff
@@ -314,25 +314,29 @@ Response body:
...
-* "nics", a list of nics, each represted by a JSON object having
-    at least the following fields:
-        * "label", the nic's label.
-        * "macaddr", the nic's mac address.
+Returns a JSON object representing a node.
+The object will have at least the following fields:
 
-Response body:
+        * "name", the name/label of the node (string).
+        * "project", the name of the project a node bleongs to or null if the node does not belong to a project
```
Comment (SahilTikale) [inline docs/rest_api.md]: typo 'bleongs'

File: haas/api.py [hunk 11]
```diff
@@ -773,32 +773,39 @@ def list_project_networks(project):
...
+    """Show the details of a node.
 
     Returns a JSON object representing a node.
+
     The object will have at least the following fields:
+
         * "name", the name/label of the node (string).
-        * "free", indicates whether the node is free or has been allocated
-            to a project.
+        * "project", the name of the project a node bleongs to or null if the node does not belong to a project
```
Comment (SahilTikale) [inline haas/api.py]: same typo

**Comment by SahilTikale** [comment]
all looks good to me. Just fix the typo and I will approve it.

**Comment by SahilTikale** [comment]
+1

**Comment by henn** [comment]
+1, merging.

**Pull request merged**